### PR TITLE
Configure Surefire to run Junit4 and Junit5 tests

### PR DESCRIPTION
**Pull request by onobc**
Several modules were configured to only run Junit 4 tests. As such, many tests were out-of-date. This commit adjusts to run both Junit 4 and 5 test as well as fix any broken/bit-rot tests.

### CODE REVIEW

File: spring-cloud-dataflow-server-core/pom.xml [hunk 1]
```diff
@@ -183,11 +178,6 @@
 			<groupId>org.springframework.cloud</groupId>
 			<artifactId>spring-cloud-task-batch</artifactId>
 		</dependency>
-		<dependency>
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/pom.xml:186]: I see we still use it, are we picking this up transitively from somewhere else?

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/db/migration/MySQL57SmokeTest.java [hunk 2]
```diff
@@ -26,6 +27,7 @@
  *
  * @author Corneil du Plessis
  */
+@Disabled("Will fix once PR is merged to run all tests")
```
Comment (cppwfs) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/db/migration/MySQL57SmokeTest.java:30]: Assumed it is this PR that is merged before activating the smoke tests.

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/AppRegistryControllerTests.java [hunk 3]
```diff
@@ -425,15 +425,15 @@ public void testListSingleApplication() throws Exception {
...
+				.andExpect(jsonPath("$.options[*]", hasSize(7)));
 	}
 
 	@Test
 	public void testListSingleApplicationExhaustive() throws Exception {
 		mockMvc.perform(get("/apps/source/time?exhaustive=true").accept(MediaType.APPLICATION_JSON))
 				.andExpect(status().isOk()).andExpect(jsonPath("name", is("time")))
 				.andExpect(jsonPath("type", is("source")))
-				.andExpect(jsonPath("$.options[*]", hasSize(2001)));
+				.andExpect(jsonPath("$.options[*]", hasSize(2022)));
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/AppRegistryControllerTests.java:436]: Was this test not being executed and thus when the tests were updated to use the latest time or log their parameter count had changed?

**Comment by cppwfs** [comment]
LGTM Great Work!!!!

File: spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/AppRegistryControllerTests.java (hunk 3 above)
Comment (onobc) [reply spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/controller/AppRegistryControllerTests.java:436]: Exactly... The more recent app has new properties (new version of boot and poller etc..)

File: spring-cloud-dataflow-server-core/pom.xml (hunk 1 above)
Comment (onobc) [reply spring-cloud-dataflow-server-core/pom.xml:186]: `spring-boot-starter-test`

**Pull request merged**